src/docs/development/tools/hot-reload.md

```diff
@@ -310,27 +293,50 @@ supported at all. These include:
   }
   ```
 
-In these situations,
-hot reload generates a diagnostic message and fails without
-committing any changes.
+* Prior to Flutter SDK 1.17, if you modified a
```

Comment:
Do we need to document older behavior? Perhaps remove the mention of the pre-1.17 reload behavior wrt class hierarchy changes.